actions/gnusocial

```diff
@@ -0,0 +1,112 @@
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""
+Configuration helper for gnusocial
+"""
+
+import argparse
+import os
+import shutil
+import subprocess
+import sys
+
+from plinth import action_utils
+
+
+def parse_arguments():
+    """Return parsed command line arguments as dictionary."""
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers(dest='subcommand', help='Sub command')
+
+    # Setup gnusocial site
+    subparsers.add_parser('setup', help='Perform first time setup operations')
+
+    # Enable gnusocial site
+    subparsers.add_parser('enable', help='Enable ikiwiki site')
+
+    # Disable gnusocial site
+    subparsers.add_parser('disable', help='Disable ikiwiki site')
+
+    pre_install = subparsers.add_parser(
+        'pre-install',
+        help='Preseed debconf values before packages are installed.')
+    subparsers.add_parser('setup', help='Setup gnusocial configuration')
+
+    return parser.parse_args()
+
+
+def subcommand_pre_install(arguments):
+    """Preseed debconf values before packages are installed."""
+
+    config_params = [b'gnu-social gnu-social/server-ssl boolean false',
+                     b'gnu-social gnu-social/mysql/admin-pass password root',
+                     b'gnu-social gnu-social/mysql/password-confirm password root',
+                     b'gnu-social gnu-social/mysql/app-pass password root',
+                     b'gnu-social gnu-social/mysql/app-password-confirm password root',
+                     b'gnu-social gnu-social/admin-password password admin',
+                     b'gnu-social gnu-social/admin-password-again password admin',
+                     b'gnu-social gnu-social/mysql/admin-user string root',
+                     b'gnu-social gnu-social/internal/skip-preseed boolean false',
+                     b'gnu-social gnu-social/admin-email string root@localhost',
+                     b'gnu-social gnu-social/missing-db-package-error string abort',
+                     b'gnu-social gnu-social/db/dbname string gnusocial',
+                     b'gnu-social gnu-social/site-name string "GNU Social"',
+                     b'gnu-social gnu-social/dbconfig-remove boolean true',
+                     b'gnu-social gnu-social/site-profile select Community',
+                     b'gnu-social gnu-social/mysql/method string "Unix socket"',
+                     b'gnu-social gnu-social/db/app-user string gnu-social@localhost',
+                     b'gnu-social gnu-social/internal/reconfiguring boolean false',
+                     b'gnu-social gnu-social/dbconfig-install boolean true',
+                     b'gnu-social gnu-social/upgrade-backup boolean true',
+                     b'gnu-social gnu-social/upgrade-error select abort',
+                     b'gnu-social gnu-social/dbconfig-upgrade boolean true',
+                     b'gnu-social gnu-social/remote/host string localhost',
```

Comment:
A single valid domain name needs to be set.  Otherwise, GNU Social will not work. To allow for setting and changing the domain that GNU Social is configured to, we should have drop down containing all the domain names available in FreedomBox. 

1. Pick a default domain name and use it during installation.  If no domain is available, set it to localhost.
1. After installation show the configuration page with a drop down selection of all domain name available.
1. If current domain name is setup to localhost, show a warning to the user that a domain needs to be set and otherwise GNU Social will not work.
1. Show user a message after installation showing the URL with which to access GNU Social.